controllers/controllervm_controller.go

```diff
@@ -240,6 +273,39 @@ func (r *ControllerVMReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	return ctrl.Result{}, nil
 }
 
+func (r *ControllerVMReconciler) setControllerHostStatus(instance *ospdirectorv1beta1.ControllerVM, hostname string, ipaddress string) {
+	// Set status for controllerHosts
+	instance.Status.ControllerHosts[hostname] = ospdirectorv1beta1.ControllerHostStatus{
+		Hostname:  hostname,
+		IPAddress: ipaddress,
+	}
+}
+
+func (r *ControllerVMReconciler) overcloudipsetCreateOrUpdate(instance *ospdirectorv1beta1.ControllerVM) (*ospdirectorv1beta1.OvercloudIPSet, controllerutil.OperationResult, error) {
+	overcloudIPSet := &ospdirectorv1beta1.OvercloudIPSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      instance.Name, // used as the base name of server instances
+			Namespace: instance.ObjectMeta.Namespace,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, overcloudIPSet, func() error {
+		overcloudIPSet.Spec.Networks = []string{"ctlplane"} //FIXME provide way to expose more networks via CR
+		overcloudIPSet.Spec.Role = "controller"             // NOTE: will VM's always be controllers
```

Comment:
we should add the role to the CRD that we could support custom roles

Reply:
Role is in!